### PR TITLE
feature[TW27693]: Add Derived-From-To fields to rest call action/details

### PR DESCRIPTION
**Pull request by dondunne**
Change-Id: Ibd43b7b477216eea6182ef7a569775949d4f4252

### CODE REVIEW

**Review by ryanbase — DISMISSED** [review]

File: plugins/org.eclipse.osee.ats.rest/src/org/eclipse/osee/ats/rest/internal/workitem/WorkItemJsonWriter.java [hunk 1]
```diff
@@ -487,8 +487,24 @@ private static void writeDerived(AtsApi atsApi, JsonGenerator writer, IAtsTeamWo
...
-         writer.writeObjectField("AtsId", teamWf.getAtsId());
+         writer.writeObjectField("AtsId",
+            atsApi.getAttributeResolver().getSoleAttributeValue(art, AtsAttributeTypes.AtsId, ""));
          writer.writeObjectField("type", art.getArtifactType().toStringWithId());
+         writer.writeObjectField("state",
+            atsApi.getAttributeResolver().getSoleAttributeValueAsString(art, AtsAttributeTypes.CurrentStateName, ""));
+         writer.writeEndObject();
+      }
+      writer.writeEndArray();
+      writer.writeArrayFieldStart("derivedto");
```
Comment (Grumb1e) [inline plugins/org.eclipse.osee.ats.rest/src/org/eclipse/osee/ats/rest/internal/workitem/WorkItemJsonWriter.java:498]: Nit: Should this be camel case? "derivedTo"

File: plugins/org.eclipse.osee.ats.ide.integration.tests/src/org/eclipse/osee/ats/ide/integration/tests/ats/branch/DemoBranchRegressionTest.java [hunk 2]
```diff
@@ -330,6 +333,44 @@ public void testCodeTaskCreationAfterCreateReqArtToDelete() {
...
+      String url2 = String.format("ats/action/%s/details", codeTeamWf.getIdString());
+      JsonNode action2 = testActionRestCall(url2, 1);
+      JsonNode derived2 = action2.get("derivedfrom");
+      Assert.assertNotNull(derived2);
+      Assert.assertEquals(reqTeamWf.getAtsId(), derived2.get(0).get("AtsId").asText());
+
+      // Test derivedTo
+      String url3 = String.format("ats/action/%s/details", reqTeamWf.getIdString());
+      JsonNode action3 = testActionRestCall(url3, 1);
+      JsonNode derived3 = action3.get("derivedto");
```
Comment (Grumb1e) [inline plugins/org.eclipse.osee.ats.ide.integration.tests/src/org/eclipse/osee/ats/ide/integration/tests/ats/branch/DemoBranchRegressionTest.java:363]: Nit: Should this be camel case? "derivedTo"

File: plugins/org.eclipse.osee.ats.ide.integration.tests/src/org/eclipse/osee/ats/ide/integration/tests/ats/branch/DemoBranchRegressionTest.java [hunk 3]
```diff
@@ -330,6 +333,44 @@ public void testCodeTaskCreationAfterCreateReqArtToDelete() {
...
+      Assert.assertEquals(reqTeamWf.getAtsId(), derived1.get("AtsId").asText());
+      Assert.assertEquals(reqTeamWf.getIdString(), derived1.get("id").asText());
+      Assert.assertEquals(reqTeamWf.getName(), derived1.get("name").asText());
+      Assert.assertEquals(reqTeamWf.getArtifactType().toStringWithId(), derived1.get("type").asText());
+      Assert.assertEquals(reqTeamWf.getCurrentStateName(), derived1.get("state").asText());
+
+      // Test derivedFrom with id
+      String url2 = String.format("ats/action/%s/details", codeTeamWf.getIdString());
+      JsonNode action2 = testActionRestCall(url2, 1);
+      JsonNode derived2 = action2.get("derivedfrom");
```
Comment (Grumb1e) [inline plugins/org.eclipse.osee.ats.ide.integration.tests/src/org/eclipse/osee/ats/ide/integration/tests/ats/branch/DemoBranchRegressionTest.java:356]: Nit: Should this be camel case? "derivedFrom"

File: plugins/org.eclipse.osee.ats.ide.integration.tests/src/org/eclipse/osee/ats/ide/integration/tests/ats/query/AtsActionEndpointImplTest.java [hunk 4]
```diff
@@ -254,6 +254,7 @@ public void testAtsActionsDetailsRestCall() {
       Assert.assertEquals(action.get("AtsId").asText(), action.get("ats.Id").asText());
       Assert.assertFalse(Strings.isNumeric(action.get("ats.Created Date").asText()));
       Assert.assertNotNull(action.get("derivedfrom"));
```
Comment (Grumb1e) [inline plugins/org.eclipse.osee.ats.ide.integration.tests/src/org/eclipse/osee/ats/ide/integration/tests/ats/query/AtsActionEndpointImplTest.java:256]: Nit: Should this be camel case? "derivedFrom"

**Review by Grumb1e — DISMISSED** [review]

File: plugins/org.eclipse.osee.ats.rest/src/org/eclipse/osee/ats/rest/internal/workitem/WorkItemJsonWriter.java (hunk 1 above)
Comment (dondunne) [reply plugins/org.eclipse.osee.ats.rest/src/org/eclipse/osee/ats/rest/internal/workitem/WorkItemJsonWriter.java]: updated to DerivedTo to match the rest of the json

File: plugins/org.eclipse.osee.ats.ide.integration.tests/src/org/eclipse/osee/ats/ide/integration/tests/ats/branch/DemoBranchRegressionTest.java (hunk 3 above)
Comment (dondunne) [reply plugins/org.eclipse.osee.ats.ide.integration.tests/src/org/eclipse/osee/ats/ide/integration/tests/ats/branch/DemoBranchRegressionTest.java]: same

File: plugins/org.eclipse.osee.ats.ide.integration.tests/src/org/eclipse/osee/ats/ide/integration/tests/ats/branch/DemoBranchRegressionTest.java (hunk 2 above)
Comment (dondunne) [reply plugins/org.eclipse.osee.ats.ide.integration.tests/src/org/eclipse/osee/ats/ide/integration/tests/ats/branch/DemoBranchRegressionTest.java]: same

File: plugins/org.eclipse.osee.ats.ide.integration.tests/src/org/eclipse/osee/ats/ide/integration/tests/ats/query/AtsActionEndpointImplTest.java (hunk 4 above)
Comment (dondunne) [reply plugins/org.eclipse.osee.ats.ide.integration.tests/src/org/eclipse/osee/ats/ide/integration/tests/ats/query/AtsActionEndpointImplTest.java]: same

**Review by ryanbase — APPROVED** [review]

**Review by Grumb1e — APPROVED** [review]

**Pull request merged**